### PR TITLE
fix migration for multiple databases

### PR DESCRIPTION
**Pull request by igorkud90**
fix for https://github.com/divio/django-cms/issues/4798

### CODE REVIEW

**Comment by yakky** [comment]
I'm -1 on this, as this should be handled by the database router in the project

**Comment by igorkud90** [comment]
yakky, how should i fix my router? Code of my router in https://github.com/divio/django-cms/issues/4798.

**Comment by clifff** [comment]
@yakky: The Django documentation specifies [this is the responsibility of the `RunPython` code](https://docs.djangoproject.com/en/1.9/howto/writing-migrations/#data-migrations-and-multiple-databases), not the router. Accordingly, the `RunPython` operation doesn't check the `allow_migrate` method, so the Router cannot influence this.

I'm running into this problem in my application too, would love to see this merged. :+1:

**Comment by stefanw** [comment]
I ran into this and it turned out my database router still had the old pre-1.8 signature which then doesn't handle `RunPython` operations.

Compare the API for [`allow_migrate` in Django 1.7](https://docs.djangoproject.com/en/1.7/topics/db/multi-db/#allow_migrate) and [`allow_migrate` in Django 1.8](https://docs.djangoproject.com/en/1.8/topics/db/multi-db/#allow_migrate)

This is the [`RunPython` code in Django 1.7](https://github.com/django/django/blob/1.7/django/db/migrations/operations/special.py#L112-L117) and [in 1.8](https://github.com/django/django/blob/1.8/django/db/migrations/operations/special.py#L175-L181).

If you use the old signature, your database router will [not be called for `RunPython` operations](https://github.com/django/django/blob/1.8/django/db/utils.py#L337-L345).

In conclusion, starting in 1.8 the database router is able to decide on `RunPython` operations and the decision to run on the current db connection does not have to be in the migration.

**Comment by clifff** [comment]
@stefanfoulis Thanks for the heads up! I was using the old signature style and this fixed my problem.

**Comment by yakky** [comment]
@stefanw thanks for the information. Closing this PR, then. It looks a much better approach
